test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -47,43 +50,59 @@ public final class PopupMenuLocation {
     private static final int SIZE = 350;
     public static final String TEXT =
             "Long-long-long-long-long-long-long text in the item-";
+    public static final int OFFSET = 50;
     private static volatile boolean action = false;
+    private static Robot robot;
+    private static Frame frame;
+    private static Rectangle screenBounds;
+
 
     public static void main(final String[] args) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(200);
         GraphicsEnvironment ge =
                 GraphicsEnvironment.getLocalGraphicsEnvironment();
         GraphicsDevice[] sds = ge.getScreenDevices();
         for (GraphicsDevice sd : sds) {
             GraphicsConfiguration gc = sd.getDefaultConfiguration();
-            Rectangle bounds = gc.getBounds();
-            Point point = new Point(bounds.x, bounds.y);
+            screenBounds = gc.getBounds();
             Insets insets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
-            while (point.y < bounds.y + bounds.height - insets.bottom - SIZE) {
-                while (point.x
-                        < bounds.x + bounds.width - insets.right - SIZE) {
+            Point point = new Point(screenBounds.x + insets.left,
+                                    screenBounds.y + insets.top);
+            while (point.y <
+                   screenBounds.y + screenBounds.height - insets.bottom -
+                   SIZE) {
+                while (point.x <
+                       screenBounds.x + screenBounds.width - insets.right -
+                       SIZE) {
                     test(point);
-                    point.translate(bounds.width / 5, 0);
+                    point.translate(screenBounds.width / 5, 0);
                 }
-                point.setLocation(bounds.x, point.y + bounds.height / 5);
+                point.setLocation(screenBounds.x,
+                                  point.y + screenBounds.height / 5);
             }
         }
     }
 
-    private static void test(final Point tmp) throws Exception {
+    private static void test(final Point tmp) {
+        frame = new Frame();
         PopupMenu pm = new PopupMenu();
-        for (int i = 1; i < 7; i++) {
-            pm.add(TEXT + i);
-        }
-        pm.addActionListener(e -> action = true);
-        Frame frame = new Frame();
+        IntStream.rangeClosed(1, 6).forEach(i -> pm.add(TEXT + i));
+        pm.addActionListener(e -> {
+            action = true;
+            System.out.println(" Got action event " + e);
+        });
+
         try {
+            frame.setUndecorated(true);
             frame.setAlwaysOnTop(true);
             frame.setLayout(new FlowLayout());
             frame.add(pm);
             frame.pack();
             frame.setSize(SIZE, SIZE);
             frame.setVisible(true);
             frame.setLocation(tmp.x, tmp.y);
```

Comment:
```suggestion
            frame.setLocation(tmp);
            frame.setVisible(true);
```
You can pass a `Point` object to `setLocation` directly.

Setting the location before making the frame visible avoids flickering where the frame appears at 0, 0 and then moves to another location.